client/driver/rkt.go

```diff
@@ -657,15 +660,19 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
 	}
 	go h.run()
 
-	d.logger.Printf("[DEBUG] driver.rkt: retrieving network information for pod %q (UUID %s) for task %q", img, uuid, d.taskName)
-	driverNetwork, err := rktGetDriverNetwork(uuid, driverConfig.PortMap)
-	if err != nil && !pluginClient.Exited() {
-		d.logger.Printf("[WARN] driver.rkt: network status retrieval for pod %q (UUID %s) for task %q failed. Last error: %v", img, uuid, d.taskName, err)
-
-		// If a portmap was given, this turns into a fatal error
-		if len(driverConfig.PortMap) != 0 {
-			pluginClient.Kill()
-			return nil, fmt.Errorf("Trying to map ports but driver could not determine network information")
+	// Only return a driver network if *not* using host networking
+	var driverNetwork *cstructs.DriverNetwork
+	if network != "host" {
+		d.logger.Printf("[DEBUG] driver.rkt: retrieving network information for pod %q (UUID %s) for task %q", img, uuid, d.taskName)
+		driverNetwork, err = rktGetDriverNetwork(uuid, driverConfig.PortMap)
+		if err != nil && !pluginClient.Exited() {
+			d.logger.Printf("[WARN] driver.rkt: network status retrieval for pod %q (UUID %s) for task %q failed. Last error: %v", img, uuid, d.taskName, err)
+
+			// If a portmap was given, this turns into a fatal error
+			if len(driverConfig.PortMap) != 0 {
+				pluginClient.Kill()
+				return nil, fmt.Errorf("Trying to map ports but driver could not determine network information")
+			}
 		}
 	}
 
```

Comment:
I am concerned about nil pointer dereferencing in downstream code like https://github.com/hashicorp/nomad/blob/a52cb34ab8f3b61746c63b436043d9a5363f0cff/command/agent/consul/client.go#L575 that use the DriverNetwork struct, isn't that going to panic when network == "host" ?

Reply:
So I have regrets about how I implemented `DriverNetwork`, but it was designed to gracefully handle `nil`s since the vast majority of drivers don't implement it (so they return nil).

https://github.com/hashicorp/nomad/blob/730e2aa4c8265fe2382996bc0cf431d8837d8325/client/driver/driver.go#L199-L201

And if you look at each of the methods on DriverNetwork they handle `nil`: https://github.com/hashicorp/nomad/blob/730e2aa4c8265fe2382996bc0cf431d8837d8325/client/structs/structs.go#L134-L186

Definitely something worth changing/fixing in the future, but I think it's out of scope for this PR.

Reply:
Agreed, thanks for the explain.